### PR TITLE
Add ability to find req files correctly

### LINKED ISSUE #35: Handle relative paths in `-r` links
This project is awesome, thank you so much!

OK, for context, here's the structure of requirements that I have:

```
requirements.txt
reqs/base.txt
reqs/dev.txt
```

`requirements.txt` contains the following:

```
-r reqs/dev.txt
```

`reqs/dev.txt` contains the following:

```
-r base.txt
werkzeug
```

`reqs/base.txt` contains the following:

```
django
```

This is a completely valid structure, as I can run `pip install -r requirements.txt` and I get both django and werkzeug installed.  If I modify the first line of `reqs/dev.txt` to read `-r reqs/base.txt`, then I get the following error upon running `pip install -r requirements.txt`: `ERROR: Could not open requirements file: [Errno 2] No such file or directory: 'reqs/reqs/base.txt'`  So pip requires that references to other requirements files be expressed as a path relative to the file being parsed.

OK, so here's the problem.  I call `pip_api.parse_requirements('requirements.txt')`, and I get the following (excerpt of a) traceback:

```
    reqs = pip_api.parse_requirements('requirements.txt')
  File "/usr/local/lib/python3.7/site-packages/pip_api/_parse_requirements.py", line 183, in parse_requirements
    lines = "".join(_read_file(filename)).replace("\\\n", "").splitlines()
  File "/usr/local/lib/python3.7/site-packages/pip_api/_parse_requirements.py", line 47, in _read_file
    with open(filename) as f:
FileNotFoundError: [Errno 2] No such file or directory: 'base.txt'
```

So it's reading `requirements.txt`, finding the line pointing to `reqs/dev.txt`, parsing it, finding a reference to `base.txt`, and then failing because it can't find that file in the repository root next to `requirements.txt`.  

`pip_api` needs to handle file paths as being relative to the file being parsed as pip does, because setting them to be parseable by `pip_api` right now makes it so that my `pip install` command no longer works.  If you agree with my conclusion, I'd be happy to submit a PR with a fix.

### PR DESCRIPTION
**Pull request by malikoth**
Fixes #35

### CODE REVIEW

File: pip_api/_parse_requirements.py [hunk 1]
```diff
@@ -204,7 +204,7 @@ def parse_requirements(filename, options=None, include_invalid=False):
 
             elif known.requirements:
                 if known.requirements not in parsed:
-                    to_parse.add(known.requirements)
+                    to_parse.add(os.path.join(os.path.dirname(filename), known.requirements))
```
Comment (malikoth) [inline pip_api/_parse_requirements.py]: This line now contains 94 characters, and is the longest line in the file.  Is that too long?  I'll split it up if so.

File: pip_api/_parse_requirements.py [hunk 2]
```diff
@@ -204,7 +204,8 @@ def parse_requirements(filename, options=None, include_invalid=False):
 
             elif known.requirements:
                 if known.requirements not in parsed:
-                    to_parse.add(known.requirements)
+                    file_location = os.path.dirname(filename)
```
Comment (di) [inline pip_api/_parse_requirements.py]: We only need to set `file_location` once, at the beginning of this method, not every single time this loop iterates.

Also `file_directory`, `path`,  `relative_path` or just `dirname` might be better names for this value.

Comment (malikoth) [reply pip_api/_parse_requirements.py]: It seems that it should be set once for each file as it gets parsed, so I moved it to the top of the while loop.

And naming things is hard.  :P

**Comment by di** [comment]
Thanks!

**Pull request merged**